core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -203,6 +203,49 @@ public DataFrameWriter write() {
         return new DataFrameWriter(defaultWriterRegistry, this);
     }
 
+    private Table padColumnsWithMissing(boolean addColumns, final Column<?>... cols) {
+        int max = 0;
+        for (Column<?> col : columnList) {
+            max = Math.max(max, col.size());
+        }
+        for (Column<?> col : cols) {
+            max = Math.max(max, col.size());
+        }
+        for (Column<?> col : columnList) {
+            while (col.size() < max) {
+                col.appendMissing();
+            }
+        }
+        for (Column<?> col : cols) {
+            while (col.size() < max) {
+                col.appendMissing();
+            }
+        }
+        if (addColumns) {
+            addColumns(cols);
+        }
+        return this;
+    }
+
+    /**
+     * Appends missing values to columns in this table,
+     * to ensure the table is rectangular.
+     * @return this Table
+     */
+    public Table padColumnsWithMissing() {
```

Comment:
Can we come up with a shorter name?

Maybe `rectangularize`? That's what R calls it: https://www.rdocumentation.org/packages/SciencesPo/versions/1.3.7/topics/rectangularize